Add new smt backend label to regression tests

NlightNFotis opened the pull request:
This PR sets up the infrastructure to allow us to run
any tests from `regression/cbmc` with configuration targeting
the new SMT backend.

The reason for this is that we want to re-use the tests present
on that folder, so that we don't have to re-write all of these
tests to drive the new SMT backend.

The final commit marks about 50 of those tests with the new
label, so that we can we can run them with that specific profile
on CI.


<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

NlightNFotis committed:
Add a `test.pl` profile to run new tests under the new SMT backend.

This defaults to running only tests marked with an a specific label,
`new-smt-backend`. For now, no tests have been marked so, so this
new configuration runs zero tests (skips everything).

NlightNFotis committed:
Mark some of the `regression/cbmc` tests with the `new-smt-backend` label.

Now they run as part of the new `test.pl` profile added to the `CMakeLists.txt`
of the `regression/cbmc` folder, which allows us to more extensively test the
new backend, plus have the infrastructure to do so for any other test at will.

tautschnig commented on regression/cbmc/CMakeLists.txt:
```diff
@@ -24,6 +24,14 @@ add_test_pl_profile(
+  "-I;new-smt-backend"
+  "CORE"
+)
+
```
Please ensure feature-parity with the Makefile-based build system, i.e., also add this configuration in there.

NlightNFotis replied:
This has been done in the third commit (https://github.com/diffblue/cbmc/pull/6714/commits/0e1cab51823a963259378887eb1c5fb0a5be52ef)

tautschnig commented on regression/cbmc/Associativity1/test.desc:
```diff
@@ -1,4 +1,4 @@
-CORE
+CORE new-smt-backend
```
Do you perhaps want to consider adding some automation to the `check-ubuntu-20_04-cmake-gcc-KNOWNBUG` CI job in the same way that `broken-smt-backend` is exercised by that job? This would help ensure that changes that enable more tests readily are reflected in those tests.

NlightNFotis replied:
Hi Michael, this is a good idea, and I've added it in my backlog, but if done it should be part of a separate PR (primarily because I want to keep this particular PR free of changes to CI)

thomasspriggs reviewed (approved):
:+1: Looks good to me.

thomasspriggs commented on regression/cbmc/CMakeLists.txt:
```diff
@@ -24,6 +24,14 @@ add_test_pl_profile(
+add_test_pl_profile(
+  "cbmc-new-smt-backend"
+  "$<TARGET_FILE:cbmc> --incremental-smt2-solver 'z3 --smt2 -in' --slice-formula"
+  "-I;new-smt-backend"
```
We should add `-s;new-smt-backend` to this line so that separate log files get saved, rather than clobbering the log files from the standard run.

NlightNFotis replied:
Done.

thomasspriggs reviewed (approved):
The logging fix I asked for has now been addressed.

NlightNFotis committed:
Add the capacity to run specific tests against the new smt backend to the `Makefile`.

This brings our `make` support in sync with `CMake`, which already supports this.
